### ISSUE TITLE
Hitting enter on results in the find all references tree should move focus to the corresponding line in the editor

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#97817]
Issue Type: <b>Bug</b>

CC: @isidorn

1. With a keyboard, trigger a find all references in any project. Navigate to the tree view and hit enter on a result.

An editor opens showing the relevant part of the file. However, keyboard focus does not move from the tree view of results to the editor as one would expect.

VS Code version: Code - Insiders 1.46.0-insider (288852d8c26d797974bfb8aed06e1bff9b809223, 2020-05-12T05:43:31.363Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (3.61GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|33%|
</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
notes|can|1.0.0
vscode-markdownlint|Dav|0.35.1
vscode-eslint|dba|2.1.5
prettier-vscode|esb|4.7.0
rest-client|hum|0.23.2
python|ms-|2020.5.78807
remote-wsl|ms-|0.44.2
cpptools|ms-|0.28.0
Go|ms-|0.14.2
vsnotes|pat|0.7.1
rust|rus|0.7.8
markdown-preview-enhanced|shd|0.5.3
vscode-webhint|web|1.5.5

</details>

**Comment by isidorn** [comment]
Can repro. Agree. `Enter` should move focus. `Space` should preserve focus (like in the Explorer).
Assigning to @jrieken so he considers changing this

**Comment by isidorn** [comment]
@jrieken also @Neurrone just informed me he is interested in providing a PR to tackle this. So let him know if you are open for a PR in this area and if you can provide some pointers.

**Comment by jrieken** [comment]
@Neurrone Go for it. The references tree is actually an extension that lives here: https://github.com/microsoft/vscode-references-view/blob/3df68dc3f9c25c48ff215169978cd3a671311b54/src/extension.ts#L236

**Comment by Neurrone** [comment]
Thanks @jrieken. Where could I refer to to see how I should move keyboard focus to the editor?

**Comment by jrieken** [comment]
https://github.com/microsoft/vscode-references-view/blob/3df68dc3f9c25c48ff215169978cd3a671311b54/src/extension.ts#L236

**Comment by Neurrone** [comment]
How do I run Vs Code with the extension from e.g, a checked out copy of that extension instead of the built-in one? The first thing I wanted to try is to add some log statements to see what happens when pressing enter on a result in the find references tree.

**Comment by isidorn** [comment]
@Neurrone just pressing F5 to start debugging should do it. That will launch a new instance of vscode that should have the extension running ideally.

**Comment by Neurrone** [comment]
I opened that folder in Vs Code and pressed f5 when my cursor was somewhere inside the extension.ts file.

I get this notification and it fails:

```
The terminal process terminated with exit code: 1
```

Do I need to build Vs Code from source and be running Vs Code from source as well? I had some trouble with doing that with Vs Code itself - got some sort of compile error during the install stage.

**Comment by isidorn** [comment]
@Neurrone nope you do not have to run vscode out of source, you could just run regular vs code insiders or vscode stable.
If pressing F5 when debugging the extension does not work then I am not sure what is the cause. I would guess that the extension can not be build for some reason. I would try opening it in the terminal and trying to build it via `yarn`  or some other script which might be specified in package.json

I can also look into creating this PR if you continue to hit this issue.

**Comment by Neurrone** [comment]
Thanks @isidorn. I'll give this another go and will let you know if it still doesn't work.

**Comment by Neurrone** [comment]
I got it to work, turns out I forgot to do an `npm i`.

Where do I see logs being printed out, e.g via `console.log`?

**Comment by isidorn** [comment]
@Neurrone they should probably be in the debug console in the first window. `F1 > view: debug console` should show it.

**Comment by Neurrone** [comment]
Thanks, this worked.

Looking at this, the simplest way for me to fix this is to modify the `references-view.show` command being returned in provider.ts to specify `true` for the second argument of `focusEditor`. 

However, I think I understand why focus doesn't move to the editor by default. There could be situations where you just want to quickly open the file at that point without actually wanting to edit it. In that case since focus stays in the tree, you can still press the arrow keys to rapidly move between references. However, this isn't the same UX that is present everywhere else though such as in the explorer view so I don't know if there will be actual use cases affected by this.

Ideas to fix this welcome. Should I go ahead with the change or introduce a setting to allow configuring this behaviour?

In the course of working on this, I also opened #99563 and #99564.

**Comment by isidorn** [comment]
@Neurrone I do no think there should be a setting.
Like the explorer it should have two commands
`space` to open the file and not pass focus
`enter` opens the file and passes focus

**Comment by Neurrone** [comment]
@isidorn do I need to create 2 separate commands then to have both behaviour? Space is the current behaviour, and enter is new behaviour so that focus also moves?

**Comment by isidorn** [comment]
@Neurrone I think 2 commands is the way to ge here. It is I think the simplest and cleanest.

**Comment by Neurrone** [comment]
Is there any documentation somewhere on the overall structure of Vs Code? I see that there is something called a Tree Provider that I assume is magically called by the extension subsystem to populate the tree.

`getTreeItem` returns a command attached to each item:

```
result.command = { command: 'references-view.show', title: 'Open Reference', arguments: [element, true] };
```

How would I have this support 2 commands so that one can be activated by space and the other bound to enter?

**Comment by Neurrone** [comment]
@jrieken shouldn't this be a bug instead of a feature request?

**Comment by bpasero** [comment]
This should work fine now in latest, given https://github.com/microsoft/vscode/issues/85636 landed. @Neurrone can you check in insiders?

**Comment by isidorn** [comment]
Yeah this works for me nicely with latest insiders.
Let's close assing to November and @neurrone can verify
